base/utils/R/convert.input.R

```diff
@@ -99,7 +98,7 @@ convert.input <-
                      mimetype, site.id, start_date, end_date))
   
   # TODO see issue #18
-  Rbinary <- ifelse(!exists("settings") || is.null(settings$host$Rbinary),"R",settings$host$Rbinary)
+  if(!is.null(settings$host$Rbinary) Rbinary <- settings$host$Rbinary)
```

Comment:
This line can be deleted entirely -- now that `Rbinary` is an argument to the function, it's the user's responsibility to set it, and this code can simply use the value they provide (or the default of "R" if they don't specify anything); if the user wants to set it from `settings$host$Rbinary` then they should do that in their `convert.input` call.

Reply:
I 100% agree with @infotroph. Unless this line is dropped, there's still an invisible dependency on a `settings` as a global variable.

That said, it is important to check the places that convert.input is called and to add settings$host$Rbinary as an argument iif it exists in the settings object. Note that it might not exist, so don't just pass settings$host$Rbinary by default as that will result in a whole lot Rbinary = NULL errors.

Reply:
![Bleeding heart10052_square-01](https://user-images.githubusercontent.com/21123621/182022299-675a2937-5742-47d2-a4b8-101efed456a2.jpeg)